pkg/upstream/proxy.go

```diff
@@ -51,6 +51,7 @@ type multiUpstreamProxy struct {
 
 // ServerHTTP handles HTTP requests.
 func (m *multiUpstreamProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	req.URL.Path = req.URL.EscapedPath()
```

Comment:
Looking at the implementation here: https://github.com/golang/go/blob/d4b26382342c98a95b85140b2863bc30c48edd68/src/net/url/url.go#L700

If we only want to be aware of force encoded slashes resulting in a difference during the building of `URL.Path` and `URL.RawPath` we should do this I think:
```suggestion
	if req.URL.RawPath != "" {
		req.URL.Path = req.URL.EscapedPath()
	}
```
Otherwise `EscapedPath()` goes down another codepath where it will do escaping on `req.URL.Path` even though the original request builder didn't make a `RawPath` because it wasn't needed.

Reply:
how about a more generic flag switching between the sanitized go Path and the escaped raw path instead of just allowEncodedSlashes?

> And finally, we would likely need our own function that only operates on encoded slashes -- I think URL.EscapedPath() will be mutating a lot of other characters looking at the source code.
that's what i was trying to avoid. EscapedPath already considers a few edge cases and validates it. i don't wanna copy/reimplement this (and add new/other bugs).

Reply:
> how about a more generic flag switching between the sanitized go Path and the escaped raw path instead of just allowEncodedSlashes?

That seems good to me - since looking at the Go implementation, we are getting forced into more than just `%2F` auto-decoding.